modules/create_timepoint/php/ajaxtimepoint.class.inc

```diff
@@ -0,0 +1,309 @@
+            if (!$errors) {
+                try {
+                    \TimePoint::createNew(
+                        $values['identifier'],
+                        $values['subproject'],
+                        $values['visit'],
+                        isset($values['psc']) ?? null
+                    );
+                } catch (\Exception $exception) {
+                    $exception = '';
+                }
+                $response['status'] = 'success';
+            } else {
+                $response['status'] = 'error';
+                $response['errors'] = $errors;
+            }
+        } else {
+            $response['status'] = 'error';
+        }
+        return $response;
+    }
+
+    /**
+     * Initialize setup, the extra values for the
+     * create timepoint form. (psc & errors)
+     *
+     * @param array $values the user data received.
+     *
+     * @return array
+     * @throws \DatabaseException
+     * @throws \DeprecatedException
+     * @throws \LorisException
+     */
+    function initializeSetup(array $values) : array
+    {
+        // Setup variables
+        $errors         = array(); // form errors.
+        $config         =& \NDB_Config::singleton();
+        $candidate      =& \Candidate::singleton($values['identifier']);
+        $allSubprojects = \Utility::getSubprojectList();
+
+        // Frontend needs for select element.
+        $values['subproject'] = $allSubprojects;
+
+        // All subprojects from config file (error).
+        if (empty($allSubprojects)) {
+            $errors['subprojectID'] = 'No subprojects have been defined 
+            for this study. If you are an administrator, please use the 
+            Configuration module to add new subprojects.';
+        }
+        // List of valid subprojects for a given project
+        if ($config->getSetting('useProjects') === 'true') {
+            // If projects enabled, loop through and add valid subprojects only
+            $subprojList = '';
+            try {
+                $subprojList = $candidate->getValidSubprojects();
+            } catch (\DatabaseException $e) {
+            }
+            if (empty($subprojList)) {
+                $errors['subprojectID'] = 'No subprojects have been 
+                defined for the project this candidate is affiliated with. 
+                If you are an administrator, please use the Configuration module to 
+                add new subprojects and associate them with projects.';
+            }
+        }
+
+        // Retrieve visit labels.
+        $visit_options      = array();
+        $visitLabelSettings = $config->getSetting('visitLabel');
+        foreach (
+            \Utility::toArray($visitLabelSettings) as $visitLabel
+        ) {
+            if (!empty($values['subproject'])) {
+                if (isset($visitLabel['generation'])
+                    && $visitLabel['generation'] !== 'sequence'
+                ) {
+                    throw new \DeprecatedException(
```

Comment:
This has been Deprecated for a while. We should make it so that it's not possible to do this from the front-end and then delete this exception.

Reply:
@johnsaigle I'm not actually sure what visitLabel['generation'] is used for doing. Should I just delete this code?

Reply:
I think so, as long as it doesn't break anything.